### PR TITLE
[ci] add groups support

### PR DESCRIPTION
**Pull request by aslonnie**
Failed last time, let's try again.

### CODE REVIEW

**Comment by aslonnie** [comment]
test run here: https://buildkite.com/ray-project/oss-ci-build-branch/builds/4589

**Comment by aslonnie** [comment]
and with grouping: https://buildkite.com/ray-project/oss-ci-build-branch/builds/4590#0188e0d0-c5ea-4341-98ea-9fbecd743624

File: ray_ci/test_pipeline_ci.py [hunk 1]
```diff
@@ -193,5 +196,32 @@ def test_pipeline_update_queue():
     os.environ.pop("RUNNER_QUEUE_SMALL", None)
 
 
+def test_read_lineline():
+    with tempfile.TemporaryDirectory() as tmpdir:
+        pipeline_path = Path(os.path.join(tmpdir, "pipeline.yml"))
+        with open(pipeline_path, "w") as f:
+            f.write("#ci:group=foo")
```
Comment (can-anyscale) [inline ray_ci/test_pipeline_ci.py:203]: one more test where there is a group and steps is not empty?

**Review by can-anyscale — APPROVED** [review]

File: ray_ci/test_pipeline_ci.py (hunk 1 above)
Comment (aslonnie) [reply ray_ci/test_pipeline_ci.py:203]: done
